.github/workflows/ci.yml

```diff
@@ -28,27 +28,4 @@ jobs:
       - name: Vscode package plugin
         uses: JCofman/vscodeaction@master
         with:
-          args: package -o package.vsix
-
-      - uses: actions/upload-artifact@v2
-        with:
-          name: package
-          path: package.vsix
-  release:
-    if: ${{ github.ref == 'refs/heads/master' }}
-    needs: build
```

Comment:
Don't we lose the connection to the `build` step with that change? (E.g. linting might not have happened before the release.)

A different approach would be to keep everything in one file and change the condition
```
if: ${{ github.ref == 'refs/heads/master' }}
```
to
```
if: startsWith(github.ref, 'refs/tags/v')
```

as described [here](https://stackoverflow.com/questions/58475748/github-actions-how-to-check-if-current-push-has-new-tag-is-new-release).